docs/parse_argparse.py

```diff
@@ -0,0 +1,118 @@
+            wrapped=""
+            with open(path,'w') as fh_rst:
+                fh_rst.write('='*len_base + '\n' + basename + '\n' + '='*len_base + '\n' + '\n'+ '.. code-block:: shell' + '\n \n')
+                for line in stdout:
+                    wrapped = wrapped + wrapper.fill(line) + '\n' #copying the indented text
+                fh_rst.write(wrapped) #writing the indented tex to the final rst files, this will be dislayed in html
+        
+        #print(main_path)
+        main_file_no_rst = main_path + '_main' #making a file external_main
+        main_file = main_file_no_rst + '.rst' #making file external_main.rst
+        #print(main_file)
+        main_file = os.path.join(path_ori, main_file) # we now have /home/siddharth/Autometa/docs/source/scripts_2/common/external/external_main.rst
+        #print(main_file)
+        #print(file_name_base)
+
+        #this block will make the file 'script_main.rst' this is the script that will be called by the toctree in index.rst
+        if not os.path.exists('../docs/source/scripts_2/script_main.rst'):
+            with open ('../docs/source/scripts_2/script_main.rst', 'w') as fh_main:
+                fh_main.write('='*len('script_main') + '\n' + 'script_main \n' + '='*len('script_main') + '\n' + '\n'+ '.. toctree::' +
+                '\n \t :maxdepth: 3' + '\n \t :caption: Table of Contents \n')
+
+        #this will create a _main file for each directory. These rst files will be called by the toctreee in scripts_main.rst
+        if not os.path.exists(main_file):
+            len_head = len(main_path)
+            with open (main_file, 'w+') as fh_main:
+                fh_main.write('='*len_head + '\n' + main_path + '\n' + '='*len_head + '\n' + '\n'+ '.. toctree::' +
+                '\n \t :maxdepth: 2' + '\n \t :caption: Table of Contents \n' )
+
+            #need to create a link to external directory which is within the common directory
+            if main_path == 'common':
+                with open (main_file, 'a+') as fh_main:
+                    fh_main.write(f' \n \t external/external_main')
+
+            #create links to all main scripts in the scripts_main toctress, eg: external/external_main.rst (in toctree)
+            with open ('../docs/source/scripts_2/script_main.rst', 'a') as fh_main:
+                if main_path == 'external':
+                    #Common will link the external directory we do not need to include it in the main scripts_main.rst
+                    pass
+                else:
+                    fh_main.write( f'\n \t {main_path}/{main_file_no_rst}')
+
+        #open each _main.rst and add the name of the scripts within it
+        #eg in external.rst, we add samtool, bedtools, prodigal, etc
+        with open (main_file, 'a+') as fh_main:
+             fh_main.write(f'\n \t {file_name_base}')
+        
+os.remove(path_temp_write)
+os.remove(base_path_temp_write+'.rst')
```

Comment:
At its current state, this script is difficult to follow. Consider placing a few of your algorithmic decisions in functions to be utilized by the script for conciseness, readability and reusability. We can revisit after you have addressed the comments above.... 

That being said, congrats, because you have got it to work! This is the major hurdle. Now we can focus on cleaning this up while we move forwards. 👍